src/averageAges.js

```diff
@@ -54,8 +76,29 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const children = people.filter(person =>
+    onlyWithSon
+      ? person.sex === 'm' && hasMother(people, person.mother)
+      : hasMother(people, person.mother)
+  );
+
+  const ageDiffrences = children.map(child =>
+    child.born - motherBorn(people, child.mother));
+
+  const averageDiff = ageDiffrences.reduce((sum, age) => sum + age, 0);
+
+  return averageDiff / ageDiffrences.length;
+}
+
+function hasMother(people, motherName) {
+  return people.some(person => person.name === motherName);
+}
+
+function motherBorn(people, motherName) {
```

Comment:
```suggestion
function getMotherBorn(people, motherName) {
```
we name functions with verbs